tests/posix/common/src/pthread.c

```diff
@@ -402,6 +402,79 @@ void test_posix_pthread_execution(void)
 	printk("Barrier test OK\n");
 }
 
+void test_posix_pthread_error_condition(void)
+{
+	pthread_attr_t attr;
+	struct sched_param param;
+	void *stackaddr;
+	size_t stacksize;
+	int policy, detach;
+	static struct posix_thread pthread;
+	static pthread_once_t key = 1;
+
+	/* TESTPOINT: invoke pthread APIs with NULL */
+	zassert_equal(pthread_attr_destroy(NULL), EINVAL,
+		      "pthread destroy NULL error");
+	zassert_equal(pthread_attr_getschedparam(NULL, &param), EINVAL,
```

Comment:
My suggestion is to add comments for the testing purpose as possible. You don't have to write this for each zassert, but have one for the same category/scenario of the test.  Such as: 
```
        /* TESTPOINTS: Try setting attributes before init */
        ret = pthread_attr_setschedparam(&attr[0], &schedparam);
        zassert_equal(ret, EINVAL, "uninitialized attr set!");

        ret = pthread_attr_setdetachstate(&attr[0], PTHREAD_CREATE_JOINABLE);
        zassert_equal(ret, EINVAL, "uninitialized attr set!");

        ret = pthread_attr_setschedpolicy(&attr[0], schedpolicy);
        zassert_equal(ret, EINVAL, "uninitialized attr set!");
```